src/outputs/parthenon_opmd.cpp

```diff
@@ -0,0 +1,517 @@
+            mesh_comp.setPosition(std::vector<Real>{0.5, 0.5});
+            global_extent = {
+                static_cast<std::uint64_t>(pm->mesh_size.nx(X2DIR)) * effective_nx,
+                static_cast<std::uint64_t>(pm->mesh_size.nx(X1DIR)) * effective_nx,
+            };
+
+          } else {
+            PARTHENON_THROW("1D output for openpmd not yet supported.");
+          }
+          // Handling this here to now re-reset dataset later when iterating through the
+          // blocks
+          auto const dataset =
+              openPMD::Dataset(openPMD::determineDatatype<OutT>(), global_extent);
+          mesh_comp.resetDataset(dataset);
+
+          // TODO(pgrete) need unitDimension and timeOffset for this record?
+        }
+      }
+
+      // Now that the mesh record exists, actually write the data
+      auto out_var = pmb->meshblock_data.Get()->GetVarPtr(vinfo.label);
+      PARTHENON_REQUIRE_THROWS(out_var->metadata().Where() ==
+                                   MetadataFlag(Metadata::Cell),
+                               "Currently only cell centered vars are supported.");
+
+      if (out_var->IsAllocated()) {
+        // TODO(pgrete) check if we can work with a direct copy from a subview to not
+        // duplicate the memory footprint here
+#if 0        
+        // Pick a subview of the active cells of this component
+        auto const data = Kokkos::subview(
+            var->data, 0, 0, icomp, std::make_pair(kb.s, kb.e + 1),
+            std::make_pair(jb.s, jb.e + 1), std::make_pair(ib.s, ib.e + 1));
+
+        // Map a view onto a host allocation (so that we can call deep_copy)
+        auto component_buffer = buffer_list.emplace_back(ncells);
+        Kokkos::View<Real ***, Kokkos::HostSpace, Kokkos::MemoryTraits<Kokkos::Unmanaged>>
+            component_buffer_view(component_buffer.data(), nk, nj, ni);
+        Kokkos::deep_copy(component_buffer_view, data);
+#endif
+        auto out_var_h = out_var->data.GetHostMirrorAndCopy();
+        int comp_idx = 0;
+        const auto &Nt = out_var->GetDim(6);
+        const auto &Nu = out_var->GetDim(5);
+        const auto &Nv = out_var->GetDim(4);
+        // loop over all components
+        for (int t = 0; t < Nt; ++t) {
+          for (int u = 0; u < Nu; ++u) {
+            for (int v = 0; v < Nv; ++v) {
+              const auto [record_name, comp_name] =
+                  OpenPMDUtils::GetMeshRecordAndComponentNames(vinfo, comp_idx, level);
+              auto mesh_comp = it.meshes[record_name][comp_name];
+
+              const auto comp_offset = tmp_offset;
+              for (int k = kb.s; k <= kb.e; ++k) {
+                for (int j = jb.s; j <= jb.e; ++j) {
+                  for (int i = ib.s; i <= ib.e; ++i) {
+                    tmp_data[tmp_offset] = static_cast<OutT>(out_var_h(t, u, v, k, j, i));
+                    tmp_offset++;
+                  }
+                }
+              }
```

Comment:
Feels like this should be possible to merge with the higher-order functions in utils